Add [auth|security]:checkRights (#577)
kuzzleio/sdk-javascript: doc/7/controllers/auth/check-rights/index.md

---
code: true
type: page
title: checkRights
---

# checkRights

<SinceBadge version="2.8.0"/>

Checks if the provided API request can be executed by a user.

---

```js
checkRights(_id, body)
```

| Property | Type | Description |
|--- |--- |--- |

> | `_id` | <pre>string</pre> | User [kuid](/core/2/guides/main-concepts/authentication#kuzzle-user-identifier-kuid) |

Reviewer: ```suggestion
| `kuid` | <pre>string</pre> | User [kuid](/core/2/guides/main-concepts/authentication#kuzzle-user-identifier-kuid) |
```

> | `body` | <pre>object</pre> | Contains a `request` |

Reviewer: ```suggestion
| `requestPayload` | <pre>object</pre> | Contains a [RequestPayload](/core/2/api/payloads/request) |
```

---
code: true
type: page

> title: checkRights

Reviewer: ```suggestion
title: checkRights
description: Checks if an API action can be executed by the current user
```

---

# checkRights

<SinceBadge version="Kuzzle 2.8.0"/>
<SinceBadge version="auto-version"/>

> Checks if the provided API request can be executed by a user.

Reviewer: ```suggestion
Checks if the provided API request can be executed by the current logged user.
```

---

```js

> checkRights(kuid, requestPayload)

Reviewer: ```suggestion
checkRights(requestPayload)
```

```

| Property | Type | Description |

> |--- |--- |--- |
> | `kuid` | <pre>string</pre> | User [kuid](/core/2/guides/main-concepts/authentication#kuzzle-user-identifier-kuid) |

Reviewer: ```suggestion
|--- |--- |--- |
```

| `requestPayload` | <pre>object</pre> | Contains a [RequestPayload](/core/2/api/payloads/request) |

> ## Body properties

Reviewer: ```suggestion
## `requestPayload`
```
